### PR TITLE
Factor out network

### PR DESCRIPTION
**Pull request by SergiySW**
Part of #1112

### CODE REVIEW

File: nano/node/transport/transport.hpp [hunk 1]
```diff
@@ -1,10 +1,19 @@
...
 
+#include <unordered_set>
+
+#include <boost/asio/buffer.hpp>
+#include <boost/multi_index/hashed_index.hpp>
+#include <boost/multi_index/mem_fun.hpp>
+#include <boost/multi_index/member.hpp>
+#include <boost/multi_index/ordered_index.hpp>
+#include <boost/multi_index/random_access_index.hpp>
+#include <boost/multi_index_container.hpp>
```
Comment (wezrule) [inline nano/node/transport/transport.hpp]: why are these headers included? It doesn't look like a multi_index container is used in this header file.

File: nano/node/network.hpp [hunk 2]
```diff
@@ -0,0 +1,139 @@
+#pragma once
+
+#include <nano/node/common.hpp>
+#include <nano/node/stats.hpp>
```
[the remark this answers is not in the document]
Comment (SergiySW) [reply nano/node/network.hpp]: Sure

File: nano/node/transport/transport.hpp (hunk 1 above)
Comment (SergiySW) [reply nano/node/transport/transport.hpp]: It's used in tcp.hpp & udp.hpp. Can be moved to this files

Comment (SergiySW) [reply nano/node/transport/transport.hpp]: Updated

**Pull request merged**